### PR TITLE
Test feature service sendemailby csv

### PR DESCRIPTION
**Pull request by chrisbreniser**

### CODE REVIEW

File: SurveyService/src/main/java/com/revature/service/AssociateService.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.stereotype.Service;
+
+@Service 
+public interface AssociateService {
+
```
Comment (chrisbreniser) [inline SurveyService/src/main/java/com/revature/service/AssociateService.java:11]: Add comment to method

File: SurveyService/src/main/java/com/revature/service/SurveyService.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.revature.service;
+
+public interface SurveyService {
+	
+	public boolean isValidSurvey(int surveyId);
```
Comment (chrisbreniser) [inline SurveyService/src/main/java/com/revature/service/SurveyService.java:15]: comment method

File: SurveyService/src/main/java/com/revature/service/SurveySubmissionService.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.revature.service;
+
+public interface SurveySubmissionService {
+	
+	public int getSurveySubmissionByAssociateId(String batchId, int surveyId, int associateId);
+
```
Comment (chrisbreniser) [inline SurveyService/src/main/java/com/revature/service/SurveySubmissionService.java:15]: comment method

File: SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+	int associateIdAcacia;
+	int associateIdKsenia;
+	int associateIdZach;
+	
+	HashMap<String, Integer> associateIdEmailMap;
+	
+	Set<String> emails;
+	
+	Set<Integer> validAssociateIds;
```
Comment (chrisbreniser) [inline SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java]: Remove unused fields

File: SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+		
+		verify(authService).createToken(surveyId, batchId, surveySubIdAcacia);
+		verify(authService).createToken(surveyId, batchId, surveySubIdKsenia);
+		verify(authService).createToken(surveyId, batchId, surveySubIdZach);
+		
+		for(String email : emails) {
+			verify(emailService).sendEmail(url + token, email); // token used here during the test are not unique
+		}
+		
+		verify(associateService).getAssociatesByBatchId(batchId);
```
Comment (chrisbreniser) [inline SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java]: Dup verify unneeded.

File: SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+	@Test
+	void sendEmailsByCSV_withInvalidCSV() throws Exception{
+		
+		when(csvParser.parseFileForEmails(csv)).thenThrow(IllegalArgumentException.class);
+		
+		EmailResponse returned = distributionService.sendEmailsByCSV(batchId, surveyId, csv);
+		
+		verify(csvParser).parseFileForEmails(csv);
+		
+		assertEquals(returned.getSendFailedEmails().size(), 0, "Returned EmailResponse with non-empty SendFailedEmails: " + returned.getSendFailedEmails().toString());
```
Comment (chrisbreniser) [inline SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java]: should throw IllegalArgumentException and not return

File: SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+		
+		verify(csvParser).parseFileForEmails(csv);
+		
+		verify(associateService).getAssociatesByBatchId(batchId);
+		
+		assertEquals(returned.getSendFailedEmails().size(), 0, "Returned EmailResponse with non-empty SendFailedEmails: " + returned.getSendFailedEmails().toString());
+		assertEquals(returned.getTokenFailedEmails().size(), 0, "Returned EmailResponse with non-empty TokenFailedEmails: " + returned.getTokenFailedEmails().toString());
+		assertEquals(returned.getMalformedEmails().size(), 0, "Returned EmailResponse with non-empty MalformedEmails: " + returned.getMalformedEmails().toString());
+		assertEquals(returned.getStatusMessage(), "Invalid Batch Id", "Returned EmailResponse with unexpected StatusMessage: " + returned.getStatusMessage());
+
```
Comment (chrisbreniser) [inline SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java:270]: Not returning. Send InvalidSurveyIdException

**Review by chrisbreniser — COMMENTED** [review]
Fix things. Few comments left.

File: SurveyService/src/main/resources/application.yml [hunk 8]
```diff
@@ -12,14 +12,14 @@ spring:
...
-    port: ${rabbitmq_port}
+    port: ${rabbitmq_port:5672}
     username: ${rabbitmq_username}
     password: ${rabbitmq_password}
     queue-listener: ${rabbitmq_queue_listener:FormResponse-Queue}
 
   data:
     mongodb:
-      uri: ${mongodb_url}
+      uri: ${mongodb_url:mongodb+srv://test123:MongoTest123!@cluster0.fs9sm.mongodb.net/test?retryWrites=true&w=majority}
```
Comment (chrisbreniser) [inline SurveyService/src/main/resources/application.yml:22]: remove from yaml

File: SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+		try {
+			returned = distributionService.sendEmailsByCSV(batchId, surveyId, csv);
+		} catch (InvalidSurveyIdException e) {
+			fail("Threw InvalidSurveyIdException when IllegalArgumentException was exptected");
+		} catch (InvalidBatchIdException e) {
+			fail("Threw InvalidBatchIdException when IllegalArgumentException was exptected");
+		} catch (IllegalArgumentException e) {
+			// Expected path to take:
+			assertEquals(null, returned, "Returned EmailResponse with non-empty SendFailedEmails: " + returned.getSendFailedEmails().toString());
+			return;
```
Comment (chrisbreniser) [inline SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java:208]: assertThrows for each bad path.

File: SurveyService/src/main/java/com/revature/dto/AssociateDto.java [hunk 10]
```diff
@@ -0,0 +1,26 @@
+package com.revature.dto;
```
Comment (jurczakn) [inline SurveyService/src/main/java/com/revature/dto/AssociateDto.java:1]: remove from pr

File: SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+
+	/**
+	 * Test path of sendEmailsByCSV for valid parameters. Expects verification of Parsing emails from csv file,
+	 * checking formatting of each email, get valid survey submission id for each email, create token for each email,
+	 * 
+	 * @throws Exception
+	 */
+	@Test
+	void sendEmailsByCSV_withValidParameters() throws Exception{
+
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java:158]: when using this many verify methods, verify with description

File: SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java [hunk 12]
```diff
@@ -0,0 +1,273 @@
...
+	}
+	
+	/**
+	 * Test path of sendEmailsByCSV for an Invalid CSV. Expects verification of Parsing emails from csv file to fail
+	 * and handle it properly. Should throw IllegalArgumentException.
+	 * @throws Exception 
+	 */
+	@Test
+	void sendEmailsByCSV_withInvalidCSV() throws Exception{
+
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/service/DistributionServiceImplTest.java:195]: verify it doesn't try to send out emails

**Review by jurczakn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**